packages/common/src/rule-engine/operators/aml-v2-check.ts

```diff
@@ -0,0 +1,92 @@
+import { EndUserAmlHitsSchema } from '@/schemas';
+import { TWorkflowHelpers } from '@/types';
+import get from 'lodash.get';
+import isEmpty from 'lodash.isempty';
+import { z } from 'zod';
+import { ValidationFailedError } from '../errors';
+import { Rule } from '../types';
+import { OperationHelpers } from './constants';
+import { BaseOperator } from './helpers';
+import { AmlCheckV2Params } from './types';
+
+class AmlCheckV2 extends BaseOperator<any, AmlCheckV2Params> {
+  constructor() {
+    super({
+      operator: 'AML_CHECK_V2',
+    });
+  }
+
+  async extractValue(data: unknown, rule: Rule, { helpers }: { helpers: TWorkflowHelpers }) {
+    const result = z.record(z.string(), z.any()).safeParse(data);
+
+    if (!result.success) {
+      throw new ValidationFailedError('extract', 'parsing failed', result.error);
+    }
+
+    const context = result.data;
+
+    const ubosIds =
+      context?.entity?.data?.additionalInfo?.ubos?.map(
+        (ubo: { ballerineEntityId: string }) => ubo.ballerineEntityId,
+      ) ?? [];
+
+    const directorsIds =
+      context?.entity?.data?.additionalInfo?.directors?.map(
+        (director: { ballerineEntityId: string }) => director.ballerineEntityId,
+      ) ?? [];
+
+    const mainRepresentativeEndUserId =
+      context?.entity?.data?.additionalInfo?.mainRepresentative?.ballerineEntityId;
+
+    const endUsers = await Promise.all(
+      [...ubosIds, ...directorsIds, mainRepresentativeEndUserId]
+        .filter(Boolean)
+        .map(endUserId => helpers.getEndUserById(endUserId)),
+    );
+
+    if (isEmpty(endUsers)) {
```

Comment:
Consider checking before the db call if there are and ids to check, and just return.

if there are some ID's but nothing returned from the database, something is wrong and you should trigger sentry or at least a log